### ISSUE TITLE
Iteration, subdivision and training time

### ISSUE THREAD

**Issue by duanzhiihao** [DeNA/PyTorch_YOLOv3#38]
Hi, I have some confusion about how batch size and subdivision works. As described by @hirotomusiker in https://github.com/DeNA/PyTorch_YOLOv3/issues/14#issuecomment-451388804,

1 iteration = 16 batches, 1 batch = 4 images. So 100,000 iter = 1,600,000 batches = 6,400,000 images. In trainvalno5k.part there are 117,264 images, so 6,400,000 / 117,264 ~ 54.6 dataset epoches. Am I correct?

Besides, in your repo homepage, there is a nice figure that shows you trained YOLOv3 for 500,000 iterations. How long did it take to achieve that?

**Comment by hirotomusiker** [comment]
Thank you for your comment!

>In trainvalno5k.part there are 117,264 images, so 6,400,000 / 117,264 ~ 54.6 dataset epoches. Am I correct?  

If you use trainvalno5k.part, yes. We use train2017 split of COCO, which has more than 130,000 images.

>How long did it take to achieve that?  

For 500k iterations at README it takes (500k / 19.4 (iter/min) ) =  430 hours if you use V100.

**Comment by duanzhiihao** [comment]
Thank you for your reply. Besides that, could you explain why the original YOLO v3 uses SGD instead of Adam?

Based on my experiment, I tried another PyTorch implementation of YOLO v3 and find that SGD sometimes causes gradients explosion and converges slowly. But Adam works fine.

Could you kindly provide some ideas about that?

**Comment by hirotomusiker** [comment]
Most of object detection models such as Mask R-CNN use SGD instead of Adam. You would be able to see which one works better in terms of COCO AP by experiment. We can avoid gradient explosion by carefully choosing learning rate etc.
